masu/database/report_db_cleaner.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Removes report data from database."""
+
+import logging
+
+from masu.database.report_db_accessor import ReportDBAccessor
+from masu.database.reporting_common_db_accessor import ReportingCommonDBAccessor
+
+LOG = logging.getLogger(__name__)
+
+
+# pylint: disable=too-few-public-methods
+class ReportDBCleaner():
+    """Class to remove report data."""
+
+    def __init__(self, schema):
+        """Establish the database connection.
+
+        Args:
+            schema (str): The customer schema to associate with
+        """
+        self._accessor = ReportDBAccessor(schema,
+                                          ReportingCommonDBAccessor().column_map)
+
+    def purge_expired_report_data(self, expired_date):
+        """Remove report data with a billing start period before specified dat.
+
+        Args:
+            expired_date (datetime.datetime): The cutoff date for removing data.
+
+        Returns:
+            (None)
+
+        """
+        bill_objects = self._accessor.get_bill_query_before_date(expired_date)
+        for bill in bill_objects.all():
+            bill_id = bill.id
+            line_item_query = self._accessor.get_lineitem_query_for_billid(bill_id)
+            for line_item in line_item_query.all():
```

Comment:
I think the performance is going to be very bad, here.

I think we can reduce at least one level of looping by doing `line_item_query.filter(usage_start > expired_date).delete()` to let the DB do more of the heavy-lifting.

Reply:
yeah probably :). I just wanted to get enough to where something worked in unit test land for now.

So I was running into issues doing .delete() on the Query().
Might be related to this:

http://docs.sqlalchemy.org/en/latest/orm/query.html

see `Warning Additional Caveats for bulk query deletes` part under the `delete()` section

Reply:
Yep. This is likely to be a "performance critical" code segment. 

I'm okay leaving this as-is until we do a more performance-focused iteration.  It's up to you if you want to try to tackle it now or leave it for later. :smile:

Reply:
oh yeah it most definitely is.  It will surely be iterated on in the next step.  By then I will probably figure out how to do the delete on the query.

Reply:
We can come back to this but yeah, line by line deletes will be bad. We don't actually need any loops. Basically we want to extract the results of the bill query then use those results for a where clause on the delete. Mostly off the top of my head I think the delete is going to go something like `line_item_table_object.delete().where(line_item_table_object.cost_entry_bill_id == _in(bill_ids)` 

I forget how to use `IN` properly in SQLAlchemy-speak

Reply:
Yeah I figured we wouldn't ultimately want loops but I wanted some logging statements for when I start functional tests.  I was planning on coming back to tune this up once I know things are working.